src/Tablo/Tablo.tsx

```diff
@@ -0,0 +1,164 @@
+import Column, { defaultColumns, updateColumns, ColumnProps } from './Column';
+import FlexView from 'react-flexview';
+import { ColumnGroupProps } from './ColumnGroup';
+import omit = require('lodash/omit');
+
+import './patch-fixed-data-table-2';
+
+export namespace TabloProps {
+  export type SortDir = 'asc' | 'desc';
+  export type Sort<K> = {
+    sortBy?: K,
+    sortDir?: TabloProps.SortDir
+  }
+  export type ColumnChild<T, K extends string> = React.ReactElement<ColumnProps<T, K>>
+  export type ColumnGroupChild<T, K extends string> = React.ReactElement<ColumnGroupProps<T, K>>
+}
+
+export type TabloDefaultProps = {
+  /** height in pixel of every row */
+  rowHeight: number,
+  /** height in pixel of header */
+  headerHeight: number,
+  /** height in pixel of groupHeader */
+  groupHeaderHeight: number,
+  /** height in pixel of footer */
+  footerHeight: number,
+  /**  a function giving a className for a row index */
+  rowClassNameGetter: (rowIndex: number) => string | number | undefined | null,
+  /** wheater the table should resize to fit the available space. Default true. */
+  autosize: boolean
+};
+
+export type TabloRequiredProps<T, K extends string> = {
+  /** data shown in the table */
+  data: T[],
+  className?: string,
+  /** callback to be called when mouse enters a row */
+  onRowMouseEnter?: (rowIndex: number) => void,
+  /** callback to be called when mouse leaves a row */
+  onRowMouseLeave?: (rowIndex: number) => void,
+  /** an array containing the ordered list of column names, in the same order they should be rendered */
+  columnsOrder?: K[],
+  /** callback to be called when the order of columns changes after dragging an header in a new position */
+  onColumnsReorder?: (columns: K[]) => void,
+  /** callback to be called when a column is resized */
+  onColumnResize?: (x: { width: number, key: K}) => void,
+  /** table children (Column or ColumnGroup) */
+  children?: TabloProps.ColumnChild<T, K> | TabloProps.ColumnChild<T, K>[] | TabloProps.ColumnGroupChild<T, K> | TabloProps.ColumnGroupChild<T, K>[],
+  /** value of horizontal scroll */
+  scrollLeft?: number,
+  /** value of vertical scroll */
+  scrollTop?: number,
+  /** callback to be called when scrolling starts */
+  onScrollStart?: () => void,
+  /** callback to be called when scrolling ends */
+  onScrollEnd?: (x: number, y: number) => void,
+  /** the list of selected row ids */
+  selectedRows?: number[],
+  /** callback to be called when the currently selected rows change */
+  onRowsSelect?: (selectedRows: number[]) => void,
+  /** single = only one selected row at a time, multi = multiple selection, none = selection disabled */
+  selectionType?: 'single' | 'multi' | 'none',
+  /** the id of the hovered row */
+  hoveredRowIndex?: number,
+  /** callback to be called when the hovered row changes */
+  onHoverRowChange?: (rowIndex: number) => void,
+  /** id of the column according which the data should be ordered */
+  sortBy?: keyof T,
```

Comment:
yes, I encountered the same error and using `K` instead of `keyof T` fixes it (it was missing the additional keys that can be defined in `K`).
I'll fix it in a commit